modules/create_timepoint/php/ajaxtimepoint.class.inc

```diff
@@ -0,0 +1,285 @@
+                foreach ($items as $item) {
+                    $item = \Utility::toArray($item)[0];
+                    $labelOptions[$item['@']['value']] = $item['#'];
+                }
+                $visit_options[$visitLabel['@']['subprojectID']]
+                    = array_filter($labelOptions);
+            }
+        }
+        $values['visit'] = $visit_options;
+        $values['psc']   = array_intersect(
+            \Utility::getSiteList(),
+            \User::singleton()->getSiteNames()
+        );
+        if (!empty($errors)) {
+            $values['errors'] = $errors;
+        }
+        if (empty($values['psc'])) {
+            unset($values['psc']);
+        }
+
+        return new \LORIS\Http\Response\JsonResponse(
+            $values
+        );
+    }
+
+    /**
+     * Processes the values & saves to database and return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse POST request body.
+        $values = $request->getParsedBody();
+
+        // User permission verification.
+        $values = $this->verifyPermissions($request, $values);
+        if (isset($values['error'])) {
+            switch ($values['error']) {
+            case 'BadRequest':
+                return new \LORIS\Http\Response\BadRequest(
+                    'Failed to retrieve data for create timepoint.'
+                );
+            case 'Forbidden':
+                return new \LORIS\Http\Response\Forbidden(
+                    'Forbidden'
+                );
+            default:
+                break;
+            }
+        }
+
+        if (isset($values['command']) && $values['command'] == 'create') {
+            $errors = $this->validate($values);
+            // convert site entered to a \Site object
+            $site = null;
+            $user = \User::singleton();
+            $user_list_of_sites = $user->getCenterIDs();
+            $num_sites          = count($user_list_of_sites);
+            if ($num_sites == 1) {
+                $site = \Site::singleton($user_list_of_sites[0]);
+            } else if ($num_sites > 1) {
+                $site = \Site::singleton(intval($values['psc']));
+            }
+            if (!$errors && $site) {
+                \TimePoint::createNew(
+                    \Candidate::singleton($values['identifier']),
+                    intval($values['subproject']),
+                    $values['visit'],
+                    $site
+                );
+                $response['status'] = 'success';
+            } else {
+                $response['status'] = 'error';
+                $response['errors'] = $errors;
+            }
+        } else {
+            $response['status'] = 'error';
+        }
+        return new \LORIS\Http\Response\JsonResponse(
```

Comment:
(Comment/ not a formal change request) :: It would be nice if this was a 201 with the location header equal to baseurl + /candid so that the React app just need to send `window.lcoation($url)` instead of the success page with the link to click.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201

Reply:
@xlecours you want me to skip the "New time point successfully registered" and just redirect to the /candID link automatically? I can do that by just checking with the candID gets updated.